pkg/profiling/fifo_dir_test.go

```diff
@@ -0,0 +1,52 @@
+package profiling
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_fifoDir(t *testing.T) {
+	dir := t.TempDir()
+	maxFileCount := 3
+	fs := fifoDir{dirPath: dir, maxFileCount: maxFileCount}
+
+	var filesToBeDeleted []string
+	var filesToBeKept []string
+	// This should be bigger than maxFileCount to test FIFO deletion is done properly
+	numFilesToCreate := 10
+	for i := 0; i < numFilesToCreate; i++ {
+		fileName := fmt.Sprintf("%d.test.dump", i)
+		_, err := fs.Create(fileName)
+		require.NoError(t, err, "creating file: %s", fileName)
+		if i < numFilesToCreate-3 {
```

Comment:
```suggestion
		if i < numFilesToCreate-maxFileCount {
```

But personally I would try to make this test easier to read by just statically initializing `filesToBeDeleted` and `filesToBeKept` using literals, rather than generate the names in a loop, since it's so easy to have an off-by-one in such loop. Also we probably don't need to test with 10 files if limit is 3. IMHO 5 would be fine too.